Realm/RLMSyncFileManager.mm

```diff
@@ -39,7 +39,13 @@ + (NSURL *)_baseDirectory {
     dispatch_once(&onceToken, ^{
         // Create the path.
         NSFileManager *manager = [NSFileManager defaultManager];
-        NSURL *base = [NSURL fileURLWithPath:RLMDefaultDirectoryForBundleIdentifier(nil)];
+        NSString *bundleIdentifier = nil;
+        if (getenv("RLMProcessIsChild")) {
```

Comment:
`RLMProcessIsChild` is an environment variable used by tests. Non-test code should never use it.